### ISSUE TITLE
Deprecation warning logged to console

### ISSUE THREAD

**Issue by t11r** [openseadragon/openseadragon#2193]
OpenSeadragon 3.1.0 logs a warning to the browser console on initialization:

`Viewer.buttons is deprecated; Please use Viewer.buttonGroup`

This happens without any custom code and can even be seen [on the offical website](https://openseadragon.github.io/). It is not just irritating, but can mess up automatic testing of apps that include OpenSeadragon.

**Comment by pearcetm** [comment]
I've been wondering about why this happens too so I decided to dig into it. I haven't tested this yet but here's my guess as to what's going on:

It appears that it's triggered by the `$.extend()` method which was taken from jQuery long ago. While extending an object, the value is accessed:

    //openseadragon.js lines 1104-1105
    src = target[ name ]; 
    copy = options[ name ];

And, similarly, it is copied by value a few lines lower in the function.

So, the getter is called just by involving the class in extending a prototype/object, which is not ideal when using a getter that has side effects - and can cause outright bugs because setting the value returned by the getter is copied, not the getter itself. This is a known issue in jQuery's implementation that they decided not to deal with as beyond the scope of what they were trying to accomplish: https://bugs.jquery.com/ticket/6145

It would probably be better to use something like the following, in order to support extending objects with getters/setters:
```
    src = Object.getOwnPropertyDescriptor(target, name);
    copy = Object.getOwnPropertyDescriptor(options, name);
    //...
    Object.defineProperty(target, name, copy);

```

**Comment by iangilman** [comment]
Wow, good catch, and great analysis! 

Would either of you be interested in working up a fix for this?

**Comment by joedf** [comment]
I'm not sure I fully understand, but it's seems just changing `copy = options[ name ];` to `copy = Object.getOwnPropertyDescriptor(options, name).value;` removes the warning for me.
Anything else, gives the "too much recursion" error, or various errors and warnings.
I've made a pull request if no one's against this change. #2201

**Comment by iangilman** [comment]
Fixed by #2201.
